robottelo/vm.py

```diff
@@ -87,6 +87,8 @@ def __init__(
         self.ip_addr = None
         self._created = False
         self._target_image = str(id(self))
+        if tag is not None:
+            self._target_image = tag + self._target_image
```

Comment:
this will throw an exception in case `tag` is not None and is of different type than `str`.
How about:
```python
if type(tag) is str:
    self._target_image = tag + self._target_image
```